### PR TITLE
fix ExListView group header collapse-expand button

### LINKED ISSUE #4886: Can't collapse diff with parent in diff panel
*bug*

When viewing the diff panel on a merge commit, can't collapse diff with parent.
Clicking the collapse button has no effect.

clipboard01

- GitExtensions version: 2.51.01
- GIT version: 2.12.2.windows.2
- OS version: Windows 10 1607 Build 14393.1198

### LINKED ISSUE #5332: Diff list: Collapse indicator of headings not clickable and can be invisible
**Do you want to request a *feature* or report a *bug*?**
ux improvement

**What is the current behavior?**
The heading of a diff list can be double clicked what will collapse/expand the list.
But neither a single nor a double click on the collapse indicator have an effect.
grafik
The collapse indicator can become invisible.
grafik

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**
The diff list should be collapsed/expanded by single clicking the collapse indicator.
The collapse indicator should always be visible.
grafik

**Environment you encounter the issue:**

- GitExtensions version: master
- GIT version: 2.17.1.w.2-x64
- OS version: Windows 7
- .NET version: 4.7

**Did this work in previous version of GitExtensions (which)?**
have not checked yet

### LINKED ISSUE #5427: Expand / collapse button of ExListView is not working
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
Despite #5093 collapse / expand button in group header is not working neither in Dashboard nor FileStatusList. collapse / expand by double-clicking group header works though.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

1a. Click collapse button in any group header in FileStatusList. Nothing happens.

image

1b. Click collapse button in any group header in Dashboard. Nothing happens.

image

**What is the expected behavior?**

The group collapses.

**Environment you encounter the issue:**

Windows 7

### PR DESCRIPTION
**Pull request by NikolayXHD**
Fixes #5427
Fixes #5332 (not clickable part)
Fixes #4886

image

Has been tested on
- Windows 7

### CODE REVIEW

File: GitUI/UserControls/ExListView.cs [hunk 1]
```diff
@@ -219,90 +225,98 @@ public struct LVGROUP
...
                 }
 
-                return false;
+                return eventArgs.Handled;
+            }
+
+            bool IsCustomDraw(Message msg)
+            {
+                var nmhdr = (NativeMethods.NMHDR)msg.GetLParam(typeof(NativeMethods.NMHDR));
+                bool isCustomDraw = nmhdr.code == -12;
```
Comment (RussKie) [inline GitUI/UserControls/ExListView.cs]: what is this magic "-12" number?

**Comment by NikolayXHD** [comment]
@RussKie I successfully reproduced both issues on current master build.
Just replace click on expand / collapse button by dobule click on group header.

To be more specific:
1. Set `truncate long filename` setting to `None`
2. resize file status panel to the size when horizontal scroll appears
3. move scroll to rightmost position
4. double-click group header twice to collapse and re-expand the group
5. resize the panel in 1 large movement, larger than the missing width created on step 2. 

Result: both issues 1, 2 reproduced

image

**Comment by RussKie** [comment]
Hmm, I stand corrected, thank you for checking it. 
I'll raise new issues.

**Comment by RussKie** [comment]
Please rebase and I'll get it merged. Thank you

File: GitUI/UserControls/ExListView.cs (hunk 1 above)
Comment (RussKie) [reply GitUI/UserControls/ExListView.cs]: Great, thank you

**Comment by NikolayXHD** [comment]
rebase done

**Pull request merged**